frontend/public/components/environment.jsx

```diff
@@ -210,9 +243,24 @@ export const EnvironmentPage = connect(stateToProps)(
             op = 'replace';
           }
         }
-        return {path, op, value: this._envVarsToNameVal(finalPairsForContainer)};
+        return {path, op, value: this._envVarsToNameVal(finalPairsForContainer[EnvType.ENV])};
       });
-      const promise = k8sPatch(model, obj, patch);
+
+      const envFromPatch = currentEnvVars.map((finalPairsForContainer, i) => {
+        let op = 'add';
+        const path = _.isArray(rawEnvData) ? `/${envPath.join('/')}/${i}/envFrom` : `/${envPath.join('/')}/envFrom`;
+        if (_.isArray(rawEnvData)) {
+          if (rawEnvData[i].envFrom) {
+            op = 'replace';
```

Comment:
I think `'add'` will also replace existing values, so we might not need these checks. Can you try?

If this is true, we could simplify the existing code for `env` as well.